### DISCUSSION TITLE
Nextjs docs claim state is not lost between navigation? (Client side cache)

### DISCUSSION THREAD

**Discussion by gmoniava** [vercel/next.js#58540] [App Router]
In the Nextjs App router [docs][1] about Client side cache it says:

- No full-page reload between navigations and **React state and browser state is preserved**.

I can't reproduce it. See [here][2].
- Click "First Link" button, increase counter
- Click "Second Link" button
- Click "Back", to go back where the "First Link" took you
- See the counter state is 0 again

Why was the counter reset? According to above quote wasn't the counter supposed to retain value?

---
Ps there is similar quote from another section of [docs](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#4-soft-navigation) which seems to imply same thing:

> By default, the browser performs a hard navigation between pages. This means the browser reloads the page and resets React state such as useState hooks in your app and browser state such as the user's scroll position or focused element. However, in Next.js, the App Router uses soft navigation. This means React only renders the segments that have changed while preserving React and browser state, and there is no full page reload.

 

  [1]: https://nextjs.org/docs/app/building-your-application/caching#router-cache
  [2]: https://stackblitz.com/edit/stackblitz-starters-tutnca?description=The%20React%20framework%20for%20production&file=app%2Flayout.tsx&title=Next.js%20Starter

**Comment by tehandyb** [comment]
Adding to this, on my local development environment when using the PAGE based router it DOES preserve state between pages. But when deployed to a production-like environment it doesn't preserve state, and does full mounting of components with new state. Weird behavior
